cupy/random/cupy_distributions.cu

```diff
@@ -299,6 +305,175 @@ __device__ uint64_t rk_interval_64(rk_state* state, uint64_t  mx, uint64_t mask)
     return sampled;
 }
 
+__device__ int64_t rk_binomial_btpe(rk_state *state, long n, double p) {
+    double r,q,fm,p1,xm,xl,xr,c,laml,lamr,p2,p3,p4;
+    double a,u,v,s,F,rho,t,A,nrq,x1,x2,f1,f2,z,z2,w,w2,x;
+    int m,y,k,i;
+    if (!(state->has_binomial) ||
+         (state->nsave != n) ||
+         (state->psave != p)) {
+        /* initialize */
+        state->nsave = n;
+        state->psave = p;
+        state->has_binomial = 1;
+        state->r = r = min(p, 1.0-p);
+        state->q = q = 1.0 - r;
+        state->fm = fm = n*r+r;
+        state->m = m = (long)floor(state->fm);
+        state->p1 = p1 = floor(2.195*sqrt(n*r*q)-4.6*q) + 0.5;
+        state->xm = xm = m + 0.5;
+        state->xl = xl = xm - p1;
+        state->xr = xr = xm + p1;
+        state->c = c = 0.134 + 20.5/(15.3 + m);
+        a = (fm - xl)/(fm-xl*r);
+        state->laml = laml = a*(1.0 + a/2.0);
+        a = (xr - fm)/(xr*q);
+        state->lamr = lamr = a*(1.0 + a/2.0);
+        state->p2 = p2 = p1*(1.0 + 2.0*c);
+        state->p3 = p3 = p2 + c/laml;
+        state->p4 = p4 = p3 + c/lamr;
+    } else {
+        r = state->r;
+        q = state->q;
+        fm = state->fm;
+        m = state->m;
+        p1 = state->p1;
+        xm = state->xm;
+        xl = state->xl;
+        xr = state->xr;
+        c = state->c;
+        laml = state->laml;
+        lamr = state->lamr;
+        p2 = state->p2;
+        p3 = state->p3;
+        p4 = state->p4;
+    }
+  /* sigh ... */
+  Step10:
```

Comment:
could this be rewritten to use inline functions?

Reply:
AFAIK, we cannot use inline functions here as both are different variables one is a field of the struct and other is a normal variable